File: frontend-mobile/chat2date/lib/models/report_request.dart
import 'dart:convert';

class ReportRequest {
  final String userId;
  final String targetUserId;
  final String reason;
  final String? anotherReason;
  final String? description;

  ReportRequest({
    required this.userId,
    required this.targetUserId,
    required this.reason,
    this.anotherReason,
    this.description,
  });

  Map<String, dynamic> toJson() {
    return {
      'userId': userId,
      'targetUserId': targetUserId,
      'reason': reason,
      if (anotherReason != null) 'anotherReason': anotherReason,
      if (description != null) 'description': description,
    };
  }

  String toJsonString() => jsonEncode(toJson());
}

File: backend/cp25ssi2/src/main/java/sit/chat2date/cp25ssi2/dto/ReportRequest.java
package sit.chat2date.cp25ssi2.dto;

import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ReportRequest {
    @NotBlank(message = "userId is required")
    private String userId;

    @NotBlank(message = "targetUserId is required")
    private String targetUserId;

    @NotBlank(message = "reason is required")
    private String reason;

    private String anotherReason;

    private String description;
}
